apps/studio/src/features/editing-experience/components/form-builder/FormBuilder.tsx

```diff
@@ -53,32 +56,39 @@ const renderers: JsonFormsRendererRegistryEntry[] = [
 ]
 
 export interface FormBuilderProps {
-  component: keyof typeof IsomerSchema.components.complex
+  component: IsomerComplexComponentProps['type']
 }
 
 export default function FormBuilder({
   component,
 }: FormBuilderProps): JSX.Element {
-  const { properties, ...rest } = IsomerSchema.components.complex[component]
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
-  const { type, ...props } = properties
-  const schema = {
-    ...rest,
-    properties: props,
-    components: IsomerSchema.components,
-  }
-
+  const [jsonSchema, setJsonSchema] = useState<IsomerJsonSchema | null>(null)
   const [formData, setFormData] = useState({})
 
+  trpc.page.getIsomerJsonSchema.useQuery(undefined, {
```

Comment:
is it possible to cache this schema somehow? at least for majority of CMS users, seems like we fetch the same schema again and again